subsys/settings/include/settings/settings_zms.h

```diff
@@ -71,6 +80,14 @@ struct settings_zms {
 	struct settings_store cf_store;
 	struct zms_fs cf_zms;
 	const struct device *flash_dev;
+#if CONFIG_SETTINGS_ZMS_NAME_CACHE
+	struct {
+		uint32_t name_hash;
+		uint8_t flags;
```

Comment:
You could save 3B per entry if you split this array of structs into two arrays. Sounds like easy 384B RAM saving.